src/main/java/dev/sigstore/http/HttpProvider.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2022 The Sigstore Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package dev.sigstore.http;
+
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.http.apache.v2.ApacheHttpTransport;
+import java.util.concurrent.TimeUnit;
+import org.apache.http.conn.ssl.NoopHostnameVerifier;
+import org.apache.http.impl.client.HttpClientBuilder;
+
+public class HttpProvider {
```

Comment:
I'm not sure but I thought the Http stuff was an implementation detail in the chat we had. Does it need to be public?

Reply:
This is supposed to be the wrapper around the implementation. So we can swap out the underlying implementation later if we want.

Reply:
I kind of changed this around in #8

Reply:
Ideally we do something that lets the user inject their http client of choice. But that's definitely for another day.